packages/core/src/content/content-client.ts

```diff
@@ -129,4 +136,89 @@ export class ContentClient {
     const response = await this.get<LocalesQueryResponse>(GET_LOCALES_QUERY);
     return response.manyLocale;
   }
+
+  /**
+   * Retrieves all available taxonomies with optional pagination support.
+   * Note: Only taxonomies are paginated. Terms within each taxonomy are returned in full (no term pagination).
+   * @param {object} [options] - Optional pagination options.
+   * @param {number} [options.pageSize] - Optional. Limits the number of taxonomies returned per page. If not provided, defaults to Content Services API default (20).
+   * @param {string} [options.after] - Optional. Cursor for pagination. Used to fetch the next page of taxonomies.
+   * @returns A promise that resolves to a paginated list of taxonomies, including pagination metadata (`hasMore`, `cursor`) for developer-controlled pagination.
+   */
+  async getTaxonomies(options?: {
+    pageSize?: number;
+    after?: string;
+  }): Promise<ContentItemList<Taxonomy>> {
+    debug.content(
+      'Fetching taxonomies (pageSize: %s, after: %s)',
+      options?.pageSize ?? 'API Default',
+      options?.after ?? ''
+    );
+
+    const variables: Record<string, any> = {};
+    if (options?.pageSize !== undefined) variables.pageSize = options.pageSize;
+
+    variables.after = options?.after ?? '';
```

Comment:
Can we set props in variable, just for simplfication?
Can we set `pageSize` without checking for `undefined`?
```ts
    const variables: Record<string, any> = {
      pageSize: options?.pageSize,
      after: options?.after ?? '',
    };
```